File: frontend/src/api/field-config.ts
import request from './index'
import type { PageResult } from './index'

// 字段配置请求参数
export interface FieldConfigReq {
  page?: number
  pageSize?: number
  keyword?: string
  enabled?: boolean
}

// 字段配置响应
export interface FieldConfig {
  id: number
  name: string
  code: string
  type: 'text' | 'number' | 'select' | 'date' | 'datetime'
  required: boolean
  options: string[]
  defaultValue: string
  placeholder: string
  sort: number
  enabled: boolean
  createdAt: string
  updatedAt: string
}

// 创建字段配置请求
export interface CreateFieldConfigReq {
  name: string
  code: string
  type: 'text' | 'number' | 'select' | 'date' | 'datetime'
  required?: boolean
  options?: string[]
  defaultValue?: string
  placeholder?: string
  sort?: number
  enabled?: boolean
}

// 更新字段配置请求
export interface UpdateFieldConfigReq {
  id: number
  name: string
  code: string
  type: 'text' | 'number' | 'select' | 'date' | 'datetime'
  required?: boolean
  options?: string[]
  defaultValue?: string
  placeholder?: string
  sort?: number
  enabled?: boolean
}

// API 命名空间
export namespace FieldConfigApi {
  // 获取字段配置列表
  export const list = (params?: FieldConfigReq) => {
    return request.get<PageResult<FieldConfig[]>>('/field-configs', params)
  }

  // 创建字段配置
  export const create = (data: CreateFieldConfigReq) => {
    return request.post<FieldConfig>('/field-configs', data)
  }

  // 获取字段配置详情
  export const getById = (id: number) => {
    return request.get<FieldConfig>(`/field-configs/${id}`)
  }

  // 更新字段配置
  export const update = (data: UpdateFieldConfigReq) => {
    return request.put<FieldConfig>('/field-configs', data)
  }

  // 删除字段配置
  export const del = (id: number) => {
    return request.delete<null>(`/field-configs/${id}`)
  }

  // 批量删除字段配置
  export const batchDelete = (ids: number[]) => {
    return request.post<null>('/field-configs/batch-delete', { ids })
  }

  // 获取所有启用的字段配置
  export const getAllEnabled = () => {
    return request.get<FieldConfig[]>('/field-configs/all')
  }
}

File: backend/app/model/field_config.go
package model

import (
	"time"
)

// FieldConfig 字段配置
type FieldConfig struct {
	BaseModel
	Name        string `json:"name" gorm:"size:64;not null"`       // 字段名称
	Code        string `json:"code" gorm:"size:64;not null;uniqueIndex"` // 字段编码
	Type        string `json:"type" gorm:"size:32;default:text"`    // 字段类型: text/number/select/date/datetime
	Required    bool   `json:"required" gorm:"default:false"`       // 是否必填
	Options     string `json:"options" gorm:"type:text"`            // 选项(JSON数组,select类型用)
	DefaultValue string `json:"defaultValue" gorm:"size:512"`       // 默认值
	Placeholder string `json:"placeholder" gorm:"size:256"`        // 占位提示
	Sort        int    `json:"sort" gorm:"default:0"`               // 排序
	Enabled     bool   `json:"enabled" gorm:"default:true"`        // 是否启用
	IsDeleted   bool   `json:"-" gorm:"default:false"`             // 软删除
}

func (FieldConfig) TableName() string {
	return "field_configs"
}

// FieldConfigResp 字段配置响应
type FieldConfigResp struct {
	ID          uint      `json:"id"`
	Name        string    `json:"name"`
	Code        string    `json:"code"`
	Type        string    `json:"type"`
	Required    bool      `json:"required"`
	Options     []string  `json:"options"`
	DefaultValue string   `json:"defaultValue"`
	Placeholder string    `json:"placeholder"`
	Sort        int       `json:"sort"`
	Enabled     bool      `json:"enabled"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}
